dash/include/dash/allocator/CollectiveAllocator.h

```diff
@@ -123,12 +126,15 @@ class CollectiveAllocator
   /**
    * Move-assignment operator.
    */
-  self_t & operator=(const self_t && other) noexcept
+  self_t & operator=(self_t && other) noexcept
   {
     // Take ownership of other instance's allocation vector:
     if (this != &other) {
-      std::swap(_allocated, other._allocated);
```

Comment:
> This function is called in the deallocator of LocalAllocator and hence would invalidate the dart_gptr of the new LocalAllocator.

But after the assignment, the `rvalue` owns the `gptr` that were previously owned by _our_ instance of the allocator, which has no reference to them anymore after the swap. So it is OK to free them in the destructor of the `rvalue`. Anyway, your version makes the semantics more obvious :)